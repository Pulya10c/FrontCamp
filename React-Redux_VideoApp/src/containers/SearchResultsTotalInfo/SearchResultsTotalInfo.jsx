import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import InfoSubTitle from "components/InfoSubTitle";

import { videoListTotalSelector } from "selectors/videoListSelectors";

const mapStateToProps = state => ({
  total: videoListTotalSelector(state)
});

const SearchResultsTotalInfo = ({ total }) => {
  return (
    <InfoSubTitle>
      {total} {total === 1 ? "movie was" : "movies were"} found
    </InfoSubTitle>
  );
};

SearchResultsTotalInfo.propTypes = {
  total: PropTypes.number
};

export default connect(mapStateToProps, null)(SearchResultsTotalInfo);
