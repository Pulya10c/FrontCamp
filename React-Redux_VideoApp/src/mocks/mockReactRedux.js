const createMockMapStateToProps = mapStateToProps => {
  if (!mapStateToProps) {
    return mapStateToProps;
  }

  return (state, ownProps) => {
    return {
      props: mapStateToProps(state, ownProps)
    };
  };
};

const getDispatchFunctionFormShorthandObject = actionsObj => dispatch => {
  return Object.keys(actionsObj).reduce((props, actionKey) => {
    return {
      ...props,
      [actionKey]: function() {
        return dispatch(actionsObj[actionKey].apply(this, arguments));
      }
    };
  }, {});
};

const createMockMapDispatchToProps = mapDispatchToProps => {
  if (!mapDispatchToProps) {
    return mapDispatchToProps;
  }

  const mappingFunction =
    typeof mapDispatchToProps === 'function'
      ? mapDispatchToProps
      : getDispatchFunctionFormShorthandObject(mapDispatchToProps);

  return (dispatch, ownProps) => {
    dispatch = dispatch || jest.fn(action => action);

    return {
      props: mappingFunction(dispatch, ownProps),
      dispatch
    };
  };
};

const createMockMergeProps = mergeProps => {
  if (!mergeProps) {
    return mergeProps;
  }

  return (stateProps, dispatchProps, ownProps) => {
    dispatchProps = dispatchProps || { dispatch: jest.fn(action => action) };

    return {
      props: mergeProps(stateProps, dispatchProps, ownProps),
      dispatch: dispatchProps.dispatch
    };
  };
};

export default {
  connect: (
    mapStateToProps,
    mapDispatchToProps,
    mergeProps
  ) => ConnectedComponent => ({
    mapStateToProps: createMockMapStateToProps(mapStateToProps),
    mapDispatchToProps: createMockMapDispatchToProps(mapDispatchToProps),
    mergeProps: createMockMergeProps(mergeProps),
    ConnectedComponent
  })
};
