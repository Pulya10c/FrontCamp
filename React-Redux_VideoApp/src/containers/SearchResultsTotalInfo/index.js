import SearchResultsTotalInfo from './SearchResultsTotalInfo';

export default SearchResultsTotalInfo;
