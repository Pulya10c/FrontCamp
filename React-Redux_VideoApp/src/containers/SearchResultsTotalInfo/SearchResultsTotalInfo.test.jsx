import React from "react";
import { shallow } from "enzyme";
import mockReactRedux from "mocks/mockReactRedux";
import SearchResultsTotalInfo from "./";

jest.mock("react-redux", () => mockReactRedux);

const mockTotal = 42;

jest.mock("selectors/videoListSelectors", () => ({
  videoListTotalSelector: jest.fn(() => mockTotal)
}));

describe("Component SearchResultsTotalInfo", () => {
  const { mapStateToProps, ConnectedComponent } = SearchResultsTotalInfo;

  it("should render correctly", () => {
    const { props } = mapStateToProps();
    expect(shallow(<ConnectedComponent {...props} />)).toMatchSnapshot();
  });

  it("should pass correct props to component from mapStateToProps", () => {
    const { props } = mapStateToProps();
    expect(props).toEqual({
      total: mockTotal
    });
  });
});
