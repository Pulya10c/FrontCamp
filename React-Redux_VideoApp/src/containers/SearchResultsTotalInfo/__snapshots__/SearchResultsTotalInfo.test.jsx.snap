// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component SearchResultsTotalInfo should render correctly 1`] = `
<styled.h5>
  42
   
  movies were
   found
</styled.h5>
`;
